docs/hybrid_applications.md

```diff
@@ -35,39 +35,121 @@ If you are sure that you don't call the native SDK before the React Native SDK t
 
 Initialize the SDK on the native side, by using the official documentation [for iOS][2] and [for Android][3].
 
-If you have to call the native SDK before the React Native SDK and don't need the auto instrumentation for JavaScript errors, resources, or user interactions, you can initialized the SDK on the native side only.
+#### Instrumenting React Native RUM Views
 
-#### Potential issues
+On iOS, use a `UIKitRUMViewsPredicate` to filter out native views created by your navigation libraries:
 
-**You won't have auto instrumentation for JS errors, resources or interactions.**
+```swift
+class RNHybridPredicate: UIKitRUMViewsPredicate {
+    var defaultPredicate = DefaultUIKitRUMViewsPredicate()
 
-You can mitigate this by starting the auto-instrumentation by copying [what the `enableFeatures` function does][4] at the start of your React Native application.
+    func rumView(for viewController: UIViewController) -> RUMView? {
+        let canonicalClassName = NSStringFromClass(type(of: viewController))
+        // Dropping RN Views
+        if (canonicalClassName.starts(with: "RN")) {
+            return nil
+        }
 
-You must use imports to the full path to some of the files (for example, `import {DdRumErrorTracking} from '@datadog/mobile-react-native/src/rum/instrumentation/DdRumErrorTracking';`), and the path to these files might change across SDK versions:
+        return defaultPredicate.rumView(for: viewController)
+    }
+}
 
-```javascript
-import { DdRumErrorTracking } from '@datadog/mobile-react-native/src/rum/instrumentation/DdRumErrorTracking';
-import { DdRumUserInteractionTracking } from '@datadog/mobile-react-native/src/rum/instrumentation/DdRumUserInteractionTracking';
-import { DdRumResourceTracking } from '@datadog/mobile-react-native/src/rum/instrumentation/resourceTracking/DdRumResourceTracking';
-
-DdRumUserInteractionTracking.startTracking();
-DdRumResourceTracking.startTracking({
-    tracingSamplingRate: 80, // Percentage of tracing integrations for network calls between your app and your backend.
-    firstPartyHosts: ['api.example.com'] // Specify the hosts of your backends to enable tracing with these backends
-});
-DdRumErrorTracking.startTracking();
+// Use it when calling trackUIKitRUMViews
+Datadog.Configuration.trackUIKitRUMViews(using: RNHybridPredicate())
 ```
 
-### Initialize the SDK on both sides
+On Android, use a `ComponentPredicate` to filter out native views created by your navigation libraries:
+
+```kotlin
+// Adapt the Fragment type to your View tracking strategy
+class RNComponentPredicate : ComponentPredicate<Fragment> {
+    override fun accept(component: Fragment): Boolean {
+        // Identify and drop react native screen views
+        if (component.javaClass.toString().startsWith("class com.swmansion.rnscreens")) {
+            return false
+        }
+        if (component.javaClass.toString().startsWith("class com.facebook.react")) {
```

Comment:
I would suggest to use `javaClass.name` instead of `toString()`, which is more common when doing class name lookup.

Also maybe it is worth to add the comment that if reflection is used, ProGuard rules should be added to prevent obfuscation of the target class/packages in the release build, otherwise everything will be working ok in the debug build, but for the release build it will be different result.

```suggestion
        if (component.javaClass.name.startsWith("com.swmansion.rnscreens")) {
            return false
        }
        if (component.javaClass.name.startsWith("com.facebook.react")) {
```